inc/namespace.php

```diff
@@ -4,18 +4,55 @@
 
 use Exception;
 
+/**
+ * Check whether the site has a certificate set as an option.
+ *
+ * @return boolean True if the certificate is set.
+ */
 function has_certificate() : bool {
 	return (bool) get_option( 'hm-acm-certificate' );
 }
 
+/**
+ * Check whether the site's certificate has been verified.
+ *
+ * @return boolean True if the certificate is verified.
+ */
 function has_verified_certificate() {
 	return get_certificate()['Status'] === 'ISSUED';
 }
 
+/**
+ * Get the certificate details for the site.
+ *
+ * @return array An array of certificate details, derived from \AWS\Result.
+ */
 function get_certificate() : array {
 	return get_option( 'hm-acm-certificate' );
 }
 
+/**
+ * Check whether the distribution is using the linked certificate.
+ *
+ * @return bool True if certificates match, else false.
+ */
+function distribution_matches_certificate() : bool {
```

Comment:
I clicked unlink on a site that already had a certificate and then saw the following fatal error. The error persists as I refresh the page. I think it might be related to `distribution_matches_certificate()` calling `get_certificate()` as part of displaying that settings page.

So I think `distribution_matches_certificate()` needs to call `has_certificate()` and `has_cloudfront_distribution()` first to safety check against those options not existing meaning we can't verify if the distribution is using the linked cert.

We can then remove the `empty()` checks below.

![image](https://github.com/user-attachments/assets/15444021-dc6d-40c1-b1a2-6585cef9f4be)

Reply:
Ah good catch, thank you!